### PR TITLE
Sim V4L2 bridge and video driver enhancements

### PR DESCRIPTION
**Pull request by PeterBee97**
## Summary
Enhancement to the NuttX video drivers.

## Impact
Enabling NuttX simulator to capture video from the host camera (Linux v4l2 only)

## Testing
Works with some modifications on the camera app. A more comprehensive app called NXcamera will be submitted soon to the apps repository.
Virtual camera (using v4l2loopback module) is also supported.

### CODE REVIEW

File: arch/sim/Kconfig [hunk 1]
```diff
@@ -265,6 +265,29 @@ endchoice
 
 endif
 
+config SIM_VIDEO
```
Comment (xiaoxiang781216) [inline arch/sim/Kconfig:268]: @PeterBee97 let's split sim related change to new PR

Comment (PeterBee97) [reply arch/sim/Kconfig:268]: OK (well let me pass this CI first..)

**Comment by PeterBee97** [comment]
Closing, kernel changes moved to #7687

**Comment by xiaoxiang781216** [comment]
Let's reuse this PR after #7687 is merged.

**Pull request closed without merging**